### PR TITLE
Main

### PR DESCRIPTION
**Pull request by MuddyRock**
Add ability to get/set charger options.
For example, setting off peak mode, solar mode and scheduled mode all to off (0), while the charger is enabled will effectively turn  the charger on and send power to the car.

For full available options, see the class definition, and the reply from get_chargeOpts()

### CODE REVIEW

**Comment by bfayers** [comment]
Hi! thanks for the PR, just a comment or two that I'll add to individual lines

File: README.md [hunk 1]
```diff
@@ -4,6 +4,9 @@ Python Library For interacting with EO Home EV Chargers.
 
 This has only been tested with a EO Mini Pro 2.
 
+Forked by MuddyRock from @bfayers original work.
```
Comment (bfayers) [inline README.md]: Remove statement that it's a fork for this to be merged

**Review by bfayers — CHANGES_REQUESTED** [review]

**Review by bfayers — APPROVED** [review]
Thanks. I'll get around to testing the new features later on and hopefully merge this.

**Comment by MuddyRock** [comment]
Hi
Sorry - I'm a bit new to GitHub and still figuring it out (including sending messages up till now). I can see that you've approved my pull request but that the changes haven't made it into your version.
It's probably something I've done - for a start, I've now accidentally closed the PR twice!
For info: I've also worked out how to connect to the Smart Hub wifi directly. It has some open ports that might be valid targets for a man-in-the-middle intercept when the app talks to it (although the only thing I can see that the app does which require direct connect is connecting to a new WiFi network, it wouldn't surprise me if more functionality was available through this route).
Cheers
MuddyRock

**Comment by bfayers** [comment]
Nah, it's my fault -- I hadn't had a chance to test all the new features yet, but approved the changes to the code. hopefully i'll have some time over the weekend to test it.

As for the wifi network of the charger's themselves and their ports -- I've done *a lot* of digging on that already and found there is a way to put them in OCPP mode despite their claims that it doesn't support it. I've tried once to make that work but didn't really get anywhere (though, I think that was my fault) At some point I may try again to get it working on OCPP mode (that mode does completely disable functionality of the official app, though)

**Comment by MuddyRock** [comment]
No worries and no rush - it's practically Christmas after all! 
If you get five minutes at some point, I'd appreciate some info on what you can do when connected to the charger's own WiFi network (and how to do it once connected).
I have successfully MITM'd the app when it's talking via the cloud, but I can't intercept calls on the local WiFI.
In an ideal world, I'd get full local control of the charger without it going out to the cloud, but I have a feeling that's being wildly optimistic!

**Comment by bfayers** [comment]
Sorry for the long delay, got busy with stuff. All tested now and looks good to me!

**Comment by MuddyRock** [comment]
Hi 
Did you ever get anywhere with the OCPP mode?

**Comment by bfayers** [comment]
> Hi Did you ever get anywhere with the OCPP mode?

Haven't properly given it a try and frankly since the charger works at the moment I'd rather not mess with it too much haha -- I still think one of the better options is probably to turn it off, pop the charger open and get an image of the SD card (it's just a Pi0W inside!) and work towards making your own stack of software to run on it with proper OCPP support.

**Pull request merged**